mariadb/README.md

```diff
@@ -4,34 +4,114 @@
 
+## TL;DR;
+
+```bash
+$ helm install mariadb
+```
+
+## Configuration
+
+The various configuration parameters of the MariaDB container used in the Chart can be specified in the `values.yaml`.
+
+The following table briefly describes the configurable parameters of the chart.
+
+|      Parameter      |                                              Descripion                                             |
+|---------------------|-----------------------------------------------------------------------------------------------------|
+| `imageTag`            | `bitnami/mariadb` image tag.                                                                        |
+| `imagePullPolicy`     | Image pull policy. Defaults to `Always` if `imageTag` is `latest`, else defaults to `IfNotPresent`. |
+| `mariadbRootPassword` | Password for the `root` user. Defaults to `nil`.                                                    |
+| `mariadbUser`         | Username of new user to create.                                                                     |
+| `mariadbPassword`     | Password for the new user.                                                                          |
+| `mariadbDatabase`     | Name for new database to create.                                                                    |
+
+Please refer the [bitnami/mariadb](http://github.com/bitnami/bitnami-docker-mariadb) for additional information.
+
 ## Persistence
 
-For persistence of the MariaDB data, mount a [storage volume](http://kubernetes.io/docs/user-guide/volumes/) at the `/bitnami/mariadb` path of the MariaDB pod.
+For the persistence of the data and configurations a [storage volume](http://kubernetes.io/docs/user-guide/volumes/) should be mounted at the `/bitnami/mariadb` path of the MariaDB container.
 
-By default the MariaDB Chart mounts an [emptyDir](http://kubernetes.io/docs/user-guide/volumes/#emptydir) volume.
+By default, the Chart mounts an [emptyDir](http://kubernetes.io/docs/user-guide/volumes/#emptydir) volume at this location.
 
-## Configuration
+> *"An emptyDir volume is first created when a Pod is assigned to a Node, and exists as long as that Pod is running on that node. When a Pod is removed from a node for any reason, the data in the emptyDir is deleted forever."*
+
+To persist the data and configurations across Pod shutdown and startup you should swap out the `emptyDir` volume with a persistent storage volume. For the purpose of demonstration we'll use a [gcePersistentDisk](http://kubernetes.io/docs/user-guide/volumes/#gcepersistentdisk).
+
+### Step 1: Create a GCE PD using:
+
+```bash
+$ gcloud compute disks create --size=500GB --zone=us-central1-a mariadb-data-disk
+```
+
+### Step 2: Edit `templates/mariadb-deployment.yaml`
```

Comment:
I'll check if this can be automatically populated using configurations specified in the `values.yaml`. @prydonius did you happen to try anything like this before?